src/main/java/competition/subsystems/arm/commands/SetArmsToPositionCommand.java

```diff
@@ -30,9 +40,39 @@ public void initialize() {
     }
     @Override
     public void execute(){
-
+        if(pose.getAutomaticScoringInversion().get()){
+            targetPosition = automaticScoringInversion(targetPosition);
+        }
     }
     public  boolean isFinished(){
         return true;
     }
+
+    public XYPair automaticScoringInversion(UnifiedArmSubsystem.KeyArmPosition keyArm){
+        XYPair ArmAngles  = arms.getKeyArmAngles(keyArm, targetFacing)
+        Rotation2d rotationpose = pose.getCurrentFieldPose().getHeading();
+        boolean isBackwards;
+
+
+        if(pose.getAlliance() == DriverStation.Alliance.Blue && pose.isAllianceAwareField()){
+            if(rotationpose.getDegrees() > -90 && rotationpose.getDegrees() < 90){
+                isBackwards = true;
+            }
+            else{
+                isBackwards = false;
+            }
+        }
+        else{
+            if(rotationpose.getDegrees() > -90 && rotationpose.getDegrees() < 90){
+                isBackwards = false;
+            }
+            else{
+                isBackwards = true;
+            }
+        }
+        if(isBackwards){
+            ArmAngles = new XYPair(-ArmAngles.x,- ArmAngles.y);
```

Comment:
This will work for the upper arm, but not the lower arm, since it mirrors around +90 degrees. There is a mirror arm angles function in the UnifiedArm you could use instead.